src/interface/timer.rs

```diff
@@ -26,3 +29,104 @@ pub fn readtimer(now: RustInstant) -> RustDuration {
 pub fn sleep(dur: RustDuration) {
     thread::sleep(dur);
 }
+
+#[derive(Debug)]
+struct _IntervalTimer {
+    pub cageid: u64,
+    pub init_instant: RustInstant, // The instant this process is created
+    
+    pub start_instant: RustInstant,
+    pub curr_duration: RustDuration,
+    pub next_duration: RustDuration,
+
+    pub is_ticking: bool,
+}
+
+#[derive(Clone, Debug)]
+pub struct IntervalTimer {
+    _ac: Arc<Mutex<_IntervalTimer>>,
+}
+
+impl IntervalTimer {
+    pub fn new(cageid: u64) -> Self {
+        Self {
+            _ac: Arc::new(Mutex::new(
+                _IntervalTimer {
+                    cageid: cageid,
+                    init_instant: RustInstant::now(),
+                    start_instant: RustInstant::now(),
+                    curr_duration: RustDuration::ZERO,
+                    next_duration: RustDuration::ZERO,
+                    is_ticking: false,
+                }
+            ))
+        }
+    }
+
+    // Similar to getitimer. Returns (next value, current value)
+    pub fn get_itimer(&self) -> (RustDuration, RustDuration) {
+        let guard = self._ac.lock().unwrap();
```

Comment:
Should we add an error check here in case unwrap() fails?

Reply:
this unwrap would fail only when PoisonError happens, if that happens, means the system is already corrupted internally, then we would want this thread also panic isn't it?

Reply:
Yeah I think @yizhuoliang is right and this would cause a panic.